ral-facilities/datagateway
# Disable checkboxes when parent entity is in cart

## Description

This PR adds additional logic to the selection checkboxes which disables said checkboxes if the related entities parent entity is already selected. When the checkboxes are disabled, they appear greyed out and are unelectable, and a tooltip can be seen if the user hovers their mouse over the checkbox giving info as to why the checkbox is disabled.

Additional testing has been added within `selectCell.component.test.tsx` which tests the correct tooltip is displayed and that the checkbox is disabled. Despite the Codecov report indicating one line is missing coverage within [datafileTable.component.tsx](https://codecov.io/gh/ral-facilities/datagateway/pull/1497?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ral-facilities#diff-cGFja2FnZXMvZGF0YWdhdGV3YXktZGF0YXZpZXcvc3JjL3ZpZXdzL3RhYmxlL2RhdGFmaWxlVGFibGUuY29tcG9uZW50LnRzeA==), it seems to be covered fine when ran locally.

**Functionality:**
![chrome_2aNA68Xwj2](https://user-images.githubusercontent.com/52418954/211865070-694e4bda-35cf-4e54-9f01-7847eae533e2.gif)

**Tooltip:**
![image](https://user-images.githubusercontent.com/52418954/211865361-f1965e2b-9f94-4879-bc78-fa8626043a25.png)

## Testing instructions
Within `datagateway-dataview` test the following:

Investigations:
1. select investigation
2. view related datasets
3. selection checkboxes for datasets should be disabled

Datasets:
1. select dataset
2. view related datafiles
3.  selection checkboxes for datafiles should be disabled

## Agile board tracking
closes #1472
